### PR TITLE
fix: merge commits now handled

### PR DESCRIPTION
**Pull request by bcoe**
For merge commits, prEdge.node.mergeCommit.oid references the SHA of the
commit at the top of the list of commits, vs., its own SHA. We track the
SHAs observed, and if the commit references a SHA from earlier in the list
of commitHistory.edges being processed, we accept it as a valid commit.

### CODE REVIEW

File: test/fixtures/commits-with-labels.json [hunk 1]
```diff
@@ -87,7 +87,7 @@
                     "edges": [
                       {
                         "node": {
-                          "mergeCommit": {"oid": "35abf13fa8acb3988aa086f3eb23f5ce1483cc5d"},
+                          "mergeCommit": {"oid": "fcd1c890dc1526f4d62ceedad561f498195c8939"},
```
Comment (bcoe) [inline test/fixtures/commits-with-labels.json:90]: I switched the OID to an OID earlier in the list, so that we covered the edge-case addressed in this PR.
